Markdown/developer_product_guide/glossary.md

```diff
@@ -2,160 +2,132 @@
-
-### **Framework**  
-
-A framework is a pre-built, structured collection of code that provides a foundation for developing software applications. Frameworks enforce a specific architecture or methodology, promoting code organization and standardization, which makes development faster and more efficient. Unlike libraries, which the developer's code calls in its operation, a framework often calls the developer's code, defining the flow of the program.
-
-## Developer slang  
-
-The following section covers some common terms used by developers in every day usage that you may encounter from time-to-time.
+### **Wrapper**
 
-### Acronyms  
+A wrapper is code or a function that wraps around another piece of code, typically to modify or extend its behavior without altering the underlying code. Wrappers abstract complexity, add functionality (such as logging, input validation, or error handling), or simplify interaction with a complex system or library. For example, most PyAnsys libraries are Python wrappers around Ansys products. PyMAPDL wraps MAPDL in a Python interface.
 
-Developers are fond of acronyms and they get employed regularly in both their day-to-day lexicon as well as when working with best practices. A selection of the most common ones is included below.
+## Technical terms
 
-* **LGTM:** Looks good to me
+### **Dependencies**
 
-* **PEBKAC:** Problem Exists Between Keyboard And Chair
+Dependencies in programming are external libraries, modules, or packages that a software application requires to function properly. For example, PyMAPDL depends on PyVista to run. Users must download PyVista as an external package and install it separately for PyMAPDL to operate. The package manager typically handles dependency installations.
 
-* **RTFQ:** Read the *full* question
+### **Framework**
 
-* **LBYL:** Look before you leap
+A framework is a pre-built, structured collection of code that provides a foundation for developing software applications. Frameworks enforce a specific architecture or methodology, promoting code organization and standardization that makes development faster and more efficient. Unlike libraries, which the developer's code calls, a framework often calls the developer's code, defining the program's flow.
 
-* **EAFP:** Easier to ask forgiveness than permission
+### **Transient dependencies**
 
-* **DRY:** Don't Repeat Yourself
+Transient dependencies are indirect dependencies. These are the packages, modules, and libraries that a project's dependencies rely on. They are typically not declared by the package that they are needed for. For example, [PyMAPDL dependencies](https://github.com/ansys/pymapdl/blob/main/pyproject.toml) include [Matplotlib](https://matplotlib.org/stable/), but Matplotlib depends on [Pillow](https://pillow.readthedocs.io/en/latest/), which is not listed in PyMAPDL's dependencies. This means Pillow is a transient dependency.
```

Comment:
When referring to packages in descriptive text like this, it's best (and most respectful) to use the name that the creators use for the package.